src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefReturnTests.cs

```diff
@@ -1672,5 +1672,536 @@ .maxstack  1
 }");
         }
 
+        [Fact]
+        public void RefReturnConditionalAccess01()
+        {
+            var text = @"
+    using System;
+ 
+    class Program
+    {
+        class C1<T> where T : IDisposable
+        {
+            T inst = default(T);
+
+            public ref T GetDisposable()
+            {
+                return ref inst;
+            }
+
+            public void Test()
+            {
+                GetDisposable().Dispose();
+                System.Console.Write(inst.ToString());
+
+                GetDisposable()?.Dispose();
```

Comment:
We should include tests that use `?.`:

- On a constructed instance of `T` say `C1<Stream>` 
- On a constructed instance of `T` where the value is still generic: `C1<U>`